yb-voyager/cmd/importData.go

```diff
@@ -1031,17 +1028,26 @@ func getProgressAmount(filePath string) int64 {
 }
 
 func getYBSessionInitScript() []string {
-	sessionVarsPath := "/etc/yb-voyager/ybSessionVariables.sql"
-	var sessionVars []string
-	disableTransactionalWritesCmd := fmt.Sprintf("SET yb_disable_transactional_writes to %v", disableTransactionalWrites)
-	enableUpsertCmd := fmt.Sprintf("SET yb_enable_upsert_mode to %v", enableUpsert)
-	defaultSessionVars := []string{
-		"SET client_encoding to 'UTF-8'",
-		"SET session_replication_role to replica",
-		disableTransactionalWritesCmd,
-		enableUpsertCmd,
+	var defaultSessionVars []string
+
+	if checkSessionVariableSupport(SET_CLIENT_ENCODING_TO_UTF8) {
+		defaultSessionVars = append(defaultSessionVars, SET_CLIENT_ENCODING_TO_UTF8)
+	}
+	if checkSessionVariableSupport(SET_SESSION_REPLICATE_ROLE_TO_REPLICA) {
+		defaultSessionVars = append(defaultSessionVars, SET_SESSION_REPLICATE_ROLE_TO_REPLICA)
+	}
+
+	if enableUpsert {
+		if checkSessionVariableSupport(SET_YB_ENABLE_UPSERT_MODE) {
+			defaultSessionVars = append(defaultSessionVars, SET_YB_ENABLE_UPSERT_MODE)
+			// 	SET_YB_DISABLE_TRANSACTIONAL_WRITES is used only with & if upsert_mode is supported
+			if disableTransactionalWrites && checkSessionVariableSupport(SET_YB_DISABLE_TRANSACTIONAL_WRITES) {
+				defaultSessionVars = append(defaultSessionVars, SET_YB_DISABLE_TRANSACTIONAL_WRITES)
+			}
+		}
 	}
 
+	sessionVarsPath := "/etc/yb-voyager/ybSessionVariables.sql"
```

Comment:
If the file exists, then all the intelligence that we applied from line 1040 to 1047 is futile.

If the file is present, the settings from the file should be applied *in addition to* the `defaultSessionVars`.

Reply:
@amit-yb  this file is for the advanced user/use-cases having more knowledge and understanding of the database.
I think we should add any intelligence and leave it completely to the user to what all they want to set as session variables, and just simply run only statements that are provided. 
-> Basically just first check if this file exists, if true then use it otherwise our intelligent code will work.

Let me know your thoughts about this.

Reply:
@sanyamsinghal But our current installer ALWAYS creates the file! That means, with the default installation and settings, yb-voyager will never attempt upsert mode.
The `ybSessionVariables.sql` file should run *AFTER* yb-voyager sets the `defaultSessionVars`. This will allow the so called "advanced" users to set *additional* session variables.

I am having second thoughts about keeping the `ybSessionVariables.sql`. It is complicating things unnecessarily.

Reply:
Oh, i thought this file will only exists in case user creates it for their advanced usecase. I am confused a little bit here now.
My thought was more on the assumption that its full in user control of what all they want to run in session variables.

Reply:
Pushed a commit based on discussion https://github.com/yugabyte/yb-voyager/pull/220/commits/523548b675e93096af3020a08f9c108b35669077